util/dataset.go

```diff
@@ -154,7 +159,18 @@ func (d *DataSet) Validate() error {
 		return fmt.Errorf("type is not valid: %s", d.Type)
 	}
 
-	if d.IngestPipeline == "" {
+	if d.Elasticsearch != nil && d.Elasticsearch.IngestPipelineName == "" {
+		// Check that no ingest pipeline exists in the directory except default
+		for _, path := range paths {
+			if filepath.Base(path) == "default.json" || filepath.Base(path) == "default.yml" {
+				d.Elasticsearch.IngestPipelineName = "default"
+				// TODO: remove because of legacy
+				d.IngestPipeline = "default"
```

Comment:
Consider making the bare strings, `"default"`, `"default.json"`, and `"default.yml"` into constants. Especially `"default"`, since it is used in a couple of places.

Reply:
done.